Evidence/validate highlights

anathomical opened the pull request:
## WHAT
Add visual indicators to the admin tool to indicate which Activities have Highlights associated with them that don't have working matches with the associated passage.
## WHY
When a Highlight does not exactly match the associated passage, it doesn't actually work at highlighting for students.  Currently there is no easy way for Curriculum to tell when this has happened (either because of a typo, or just because the passage has been updated), so an unknown number of highlights are not working and there's no way for Curriculum to figure out which ones that is without an arduous manual audit process.  Automating the flagging of these items will make it much easier for Curriculum to get them fixed.
## HOW
- Added a set of functions to the `Highlight`, `PlagiarismText` and `Activity` models to flag cases where highlights have not valid match.
- Added the results of those functions to serialized payloads so that they are returned as part of the API responses.
- Modified the front-end to display:
- - Which activities have any highlights that are invalid in the overall activities list
- - Links to specific rules that have highlights that are invalid on the activities detail page
- - An error message on rules pages that flags specific highlights that are not valid

### Screenshots
![image](https://user-images.githubusercontent.com/331565/134603009-f137ab9c-bea0-41e3-92c9-41b6e4efb87a.png)
![image](https://user-images.githubusercontent.com/331565/134603037-88412d0b-74e6-4b1f-ba4b-a012546137ff.png)
![image](https://user-images.githubusercontent.com/331565/134603075-8500ebdc-7606-471e-bfe9-cf8db8dc0001.png)


### Notion Card Links
https://www.notion.so/quill/Implementing-Evidence-Highlighting-validation-35beda02b591499fb3ce8c06bb3590d4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Yes
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | IN PROGRESS

anathomical committed:
Update Evidence Activity and Highlight models to report validity

So that we can flag to Curriculum when there is a highlight that doesn't actually tie into a passage

anathomical committed:
Update highlight errors to include Plagiarism Text violations

anathomical committed:
Update some of the styling on Highlight validation notifications

anathomical committed:
Add a list of invalid highlights to the Activity detail view

anathomical committed:
Set width for activities list view

eadams17 commented on services/QuillLMS/client/app/bundles/Staff/components/evidence/configureSettings/activityForm.tsx:
```diff
@@ -146,6 +166,30 @@ const ActivityForm = ({ activity, handleClickArchiveActivity, requestErrors, sub
+    });
+
+    const dataTableFields = [
+      { name: "Invalid Highlights", attribute:"link", width: "100%" }
```
based off seeing that `[Object object]` tooltip in the second screenshot and the fact that all the values will be short strings like "Rule #344", you can add `noTooltip: true` to this hash to remove that

eadams17 commented on services/QuillLMS/client/app/bundles/Staff/components/evidence/configureSettings/activityForm.tsx:
```diff
@@ -253,6 +297,7 @@ const ActivityForm = ({ activity, handleClickArchiveActivity, requestErrors, sub
           handleSetPrompt={handleSetPrompt}
         />
       </form>
+      {invalid_highlights && invalid_highlights.length > 0 && renderInvalidHighlightLinks(invalid_highlights)}
```
I think it would be nice to have this bool check above the return as a variable, something like `const invalidHighlightsPresent = invalid_highlights && invalid_highlights.length > 0`

eadams17 reviewed (approved):
Looks good! just two small comments

anathomical committed:
Add prompt_id to invalid highlights payload

Since we need that value for building functional links to AutoML rules in the admin tool

anathomical committed:
Make some additional updates based on Lindsay's feedback

anathomical committed:
Remove validation logic around Plagiarism text until we figure it out

anathomical committed:
Fix linting errors

anathomical committed:
Merge branch 'develop' of github.com:empirical-org/Empirical-Core into evidence/validate-highlights

anathomical committed:
Update based on Eric feedback

anathomical committed:
Merge branch 'develop' of github.com:empirical-org/Empirical-Core into evidence/validate-highlights

anathomical committed:
Minor tweaks to link to invalid highlights for Semantic rules
